src/main.rs

```diff
@@ -188,15 +190,40 @@ async fn main() {
         tracing::error!("Failed to save playerlist: {:?}", e);
     }
 
+    let steam_user = match settings.get_steam_user() {
+        Some(user) => user,
+        None => {
+            tracing::error!("Failed to load Steam user from settings.");
+            return;
```

Comment:
We shouldn't return here. In case something goes wrong and the steam user cannot be found for some reason, there's no way around this to continue using the (still almost fully functional) app.